lib/ansible/modules/cloud/amazon/s3_sync.py

```diff
@@ -469,13 +480,11 @@ def main():
     result = {}
     mode = module.params['mode']
 
-
-    try:
-        region, ec2_url, aws_connect_kwargs = ansible.module_utils.ec2.get_aws_connection_info(module, boto3=True)
+    region, ec2_url, aws_connect_kwargs = ansible.module_utils.ec2.get_aws_connection_info(module, boto3=True)
+    if region:
         s3 = ansible.module_utils.ec2.boto3_conn(module, conn_type='client', resource='s3', region=region, endpoint=ec2_url, **aws_connect_kwargs)
-        s3.list_buckets()
-    except botocore.exceptions.NoCredentialsError as e:
-        module.fail_json(msg=str(e))
+    else:
```

Comment:
perhaps "if not region"? that keeps the standard flow from being in an "else", lets us bring the indenting back a level, etc.

Otherwise this is fantastic. Thanks for pep8 and removing the stray code.